ArduCopter/surface_tracking.cpp

```diff
@@ -1,67 +1,45 @@
 #include "Copter.h"
 
-// adjust_climb_rate - hold copter at the desired distance above the
+// set_climb_rate - hold copter at the desired distance above the
 //      ground; returns climb rate (in cm/s) which should be passed to
 //      the position controller
-float Copter::SurfaceTracking::adjust_climb_rate(float target_rate)
+float Copter::SurfaceTracking::get_surface_offset()
 {
+    float offset_cm = 0.0;
 #if RANGEFINDER_ENABLED == ENABLED
     // check tracking state and that range finders are healthy
-    if ((surface == Surface::NONE) ||
-        ((surface == Surface::GROUND) && (!copter.rangefinder_alt_ok() || (copter.rangefinder_state.glitch_count != 0))) ||
-        ((surface == Surface::CEILING) && !copter.rangefinder_up_ok()) || (copter.rangefinder_up_state.glitch_count != 0)) {
-        return target_rate;
-    }
-
-    // calculate current ekf based altitude error
-    const float current_alt_error = copter.pos_control->get_pos_target_z_cm() - copter.inertial_nav.get_altitude();
-
-    // init based on tracking direction/state
-    RangeFinderState &rf_state = (surface == Surface::GROUND) ? copter.rangefinder_state : copter.rangefinder_up_state;
-    const float dir = (surface == Surface::GROUND) ? 1.0f : -1.0f;
-
-    // reset target altitude if this controller has just been engaged
-    // target has been changed between upwards vs downwards
-    // or glitch has cleared
-    const uint32_t now = millis();
-    if ((now - last_update_ms > SURFACE_TRACKING_TIMEOUT_MS) ||
-        reset_target ||
-        (last_glitch_cleared_ms != rf_state.glitch_cleared_ms)) {
-        target_dist_cm = rf_state.alt_cm + (dir * current_alt_error);
-        reset_target = false;
-        last_glitch_cleared_ms = rf_state.glitch_cleared_ms;\
-    }
-    last_update_ms = now;
+    if (((surface == Surface::GROUND) && copter.rangefinder_alt_ok() && (copter.rangefinder_state.glitch_count == 0)) ||
```

Comment:
we should consider going back to the early return style we had in the original code and just return 0.  The early return makes it easy to quickly dismiss the trivial case instead of having it add extra complexity below.